### PR TITLE
ci: use circleci to get a preview of the generated documentation

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR is an attempt to move the static-test from travis to [circleci](https://circleci.com/). The main ideas behind are:
- use the riotbuild docker image to run the static-test and  be consistent with workers of Murdock
- have a per PR and online access to the generated documentation (this was my initial goal in fact) using the [build artifacts](https://circleci.com/docs/2.0/artifacts/) feature provided by circleci.

I tested the solution on my own fork and it works. One can see the results here: https://circleci.com/gh/aabadie/RIOT/8 and the generated documentation here: https://8-44016379-gh.circle-artifacts.com/0/tmp/doc/html/index.html

The main problem I see is the time it takes to upload the artifacts to the final report. The total build time is 9 minutes and the upload takes a bit more than 7 minutes.

### Issues/PRs references

None

### CODE REVIEW

**Comment by kaspar030** [comment]
> have a per PR and online access to the generated documentation (this was my initial goal in fact) using the build artifacts feature provided by circleci.

we could also save murdock's output.

**Comment by aabadie** [comment]
> we could also save murdock's output

I thought of it but the effort was not the same, at least for me.

**Comment by miri64** [comment]
Is there still interest in this PR?

**Comment by aabadie** [comment]
I've updated the initial goal of this PR to only build the documentation and use Circle CI Artifacts to get a preview of the generated documentation.
The last run result (on my fork) is available here: https://1526-44016379-gh.circle-artifacts.com/0/tmp/doc/html/index.html

The bad news is that now it take more than 25 minutes to upload the artifacts to S3 (the storage backend on CircleCI). This is because the doxygen documentation generates a lot of single file and they are copied one by one.
The advantage is that it requires no setup to get the preview (except the circleci config file, which is rather simple).

**Comment by miri64** [comment]
> The bad news is that now it take more than 25 minutes to upload the artifacts to S3 (the storage backend on CircleCI).

Are there other backend options?

**Comment by aabadie** [comment]
> Are there other backend options?

I don't think so: https://circleci.com/docs/2.0/artifacts/

**Comment by aabadie** [comment]
still useful, although slow.

**Comment by fjmolinas** [comment]
Can we revive this one? As long as `circle-ci` does not block a PR getting merged I find this quite useful (even if slow), I have had many times issues or question about whether my locally generated doc would match what would be displayed on `riot.org`.

**Comment by miri64** [comment]
What needs to be configured to add the status of circle-ci to the PR?

**Comment by aabadie** [comment]
> What needs to be configured to add the status of circle-ci to the PR?

I think integration of external applications, similar to Travis I guess.

If 25 minutes is acceptable before the documentation preview is available, using circleci is useful. Unfortunately, because of the huge amount of single files to upload (+19k) 99% of the time is spent in uploading the artifacts.
To speed-up things there are alternatives:
- retrieve the documentation in an archive, still using circleci : this is way faster (a couple of minutes) but no online preview.
- use github actions: there's one for uploading artifacts. I just tested in #14308 and it works, just takes 15 minutes to complete but has the same limitation: the documentation is available in an archive, so no online preview.

**Comment by miri64** [comment]
We could (and probably should), like with Travis make the results non-binding. If for any reason the artefacts are of any interest, the wait time is not much longer as with current Murdock ;-).

**Comment by miri64** [comment]
> I think integration of external applications, similar to Travis I guess.

You have some doc on that?

**Comment by aabadie** [comment]
> You have some doc on that?

Could this one help: https://circleci.com/docs/2.0/gh-bb-integration/#section=projects ?

**Comment by miri64** [comment]
Can you do a force push to see if it worked?

**Comment by miri64** [comment]
https://app.circleci.com/pipelines/github/RIOT-OS/RIOT

**Comment by aabadie** [comment]
Integration seems to work, I see a new circleci item.

**Comment by miri64** [comment]
> Integration seems to work, I see a new circleci item.

You seem to need to log in (via GitHub) to be able to see the progress.

**Comment by aabadie** [comment]
So after 27 minutes of copy... the documentation artifacts are available [here](https://app.circleci.com/pipelines/github/aabadie/RIOT/258/workflows/50d800dd-771f-4627-b30a-b91f23c573d7/jobs/2803/artifacts). And the generated documentation is available at https://2803-44016379-gh.circle-artifacts.com/0/doc/index.html

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+version: 2
+
+jobs:
```
Comment (miri64) [inline .circleci/config.yml:3]: Can you maybe add a comment in this file, that this is for doc building and that the upload takes ~30min for future reference?

Comment (aabadie) [reply .circleci/config.yml:8]: Done

Comment (miri64) [reply .circleci/config.yml:8]: Thanks!

**Pull request merged**